homeassistant/components/sensor/ambient_station.py

```diff
@@ -0,0 +1,212 @@
+"""
+Support for Ambient Weather Station Service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.ambient_station/
+"""
+
+import asyncio
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_API_KEY, CONF_MONITORED_CONDITIONS
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['ambient_api==1.5.2']
+
+CONF_APP_KEY = 'app_key'
+
+SENSOR_NAME = 0
+SENSOR_UNITS = 1
+
+CONF_UNITS = 'units'
+UNITS_US = 'us'
+UNITS_SI = 'si'
+UNIT_SYSTEM = {UNITS_US: 0, UNITS_SI: 1}
+
+SCAN_INTERVAL = timedelta(seconds=300)
+
+SENSOR_TYPES = {
+    'winddir': ['Wind Dir', 'º'],
+    'windspeedmph': ['Wind Speed', 'mph'],
+    'windgustmph': ['Wind Gust', 'mph'],
+    'maxdailygust': ['Max Gust', 'mph'],
+    'windgustdir': ['Gust Dir', 'º'],
+    'windspdmph_avg2m': ['Wind Avg 2m', 'mph'],
+    'winddir_avg2m': ['Wind Dir Avg 2m', 'mph'],
+    'windspdmph_avg10m': ['Wind Avg 10m', 'mph'],
+    'winddir_avg10m': ['Wind Dir Avg 10m', 'º'],
+    'humidity': ['Humidity', '%'],
+    'humidityin': ['Humidity In', '%'],
+    'tempf': ['Temp', ['ºF', 'ºC']],
+    'tempinf': ['Inside Temp', ['ºF', 'ºC']],
+    'battout': ['Battery', ''],
+    'hourlyrainin': ['Hourly Rain Rate', 'in/hr'],
+    'dailyrainin': ['Daily Rain', 'in'],
+    '24hourrainin': ['24 Hr Rain', 'in'],
+    'weeklyrainin': ['Weekly Rain', 'in'],
+    'monthlyrainin': ['Monthly Rain', 'in'],
+    'yearlyrainin': ['Yearly Rain', 'in'],
+    'eventrainin': ['Event Rain', 'in'],
+    'totalrainin': ['Lifetime Rain', 'in'],
+    'baromrelin': ['Rel Pressure', 'inHg'],
+    'baromabsin': ['Abs Pressure', 'inHg'],
+    'uv': ['uv', 'Index'],
+    'solarradiation': ['Solar Rad', 'W/m^2'],
+    'co2': ['co2', 'ppm'],
+    'lastRain': ['Last Rain', ''],
+    'dewPoint': ['Dew Point', ['ºF', 'ºC']],
+    'feelsLike': ['Feels Like', ['ºF', 'ºC']],
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_APP_KEY): cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Optional(CONF_UNITS): vol.In([UNITS_SI, UNITS_US]),
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Initialze each sensor platform for each monitored condition."""
+    api_key = config[CONF_API_KEY]
+    app_key = config[CONF_APP_KEY]
+    station_data = AmbientStationData(hass, api_key, app_key)
+    if station_data.connect_success:
```

Comment:
Invert the check, and if true log error and return. Then we can save the following indentation.